tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,210 @@
+                          string(credentialsId: 'RANCHER_BYO_TLS_KEY', variable: 'RANCHER_BYO_TLS_KEY'),
+                          string(credentialsId: 'RANCHER_LINODE_ACCESSKEY', variable: "RANCHER_LINODE_ACCESSKEY")]) {
+          
+        withEnv(paramsMap) {
+          stage('Checkout') {
+            deleteDir()
+            checkout([
+                      $class: 'GitSCM',
+                      branches: [[name: "*/${branch}"]],
+                      extensions: scm.extensions + [[$class: 'CleanCheckout']],
+                      userRemoteConfigs: scm.userRemoteConfigs
+                    ])
+          }
+          dir ("./") {
+              stage('Configure and Build') {
+                if (env.AWS_SSH_PEM_KEY && env.AWS_SSH_KEY_NAME) {
+                  dir("./tests/v2/validation/.ssh") {
+                    def decoded = new String(AWS_SSH_PEM_KEY.decodeBase64())
+                    writeFile file: AWS_SSH_KEY_NAME, text: decoded
+                  }
+                }
+
+                sh "./tests/v2/validation/configure.sh"
+
+              }
+              stage('Setup') {
+                
+              } 
+              stage('Run Jobs') {
+                dir("./tests/v2/validation/") {
+                  // install yq
+                  sh returnStdout: true, script: 'wget -qO ./yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64'
+                  sh returnStdout: true, script: 'chmod a+x ./yq'
+                  def amis = env.RANCHER_AMI_LIST
+                  amiList = amis.split(',').each { obj -> 
+                  String ami
+                  String sshUser
+                  if (obj.contains(':')){
+                    def data = obj.split(':')
+                    ami = data[0]
+                    sshUser = data[1]
+                  } else {
+                    ami = obj
+                    sshUser = 'ec2-user'
+                  }
+
+                  env.RKE1_NODE_CONFIG = sh (
+                  script: "./yq '.awsNodeTemplate.ami = \"$ami\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE1_NODE_CONFIG = sh (
+                  script: "./yq '.awsNodeTemplate.sshUser = \"$sshUser\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE2_NODE_CONFIG = sh (
+                  script: "./yq '.awsMachineConfig.ami = \"$ami\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE2_NODE_CONFIG = sh (
+                  script: "./yq '.awsMachineConfig.sshUser = \"$sshUser\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  // RKE1customconfig
+
+                  def rke1CustomFilename = "rke1CustomConfig.yaml"
+                  def rke1CustomConfig = env.RANCHER_RKE1_CUSTOM_CONFIG
+
+                  writeFile file: rke1CustomFilename, text: rke1CustomConfig
```

Comment:
I think you can get away with just one customConfig since both rke1 and rke2 would use the same one.

Reply:
That could work, but unless I'm missing something that means the Jenkins job would need to take the RKE1 and RKE2 versions as parameters so the single `awsEC2Config` could be changed dynamically, and the configs for RKE1 and RKE2 Custom Clusters would have to be exactly the same other than the RKE version. I don't think it ends up simplifying things, and ultimately I believe it's going to be a little easier to understand what's going on with the test if all the clusters work the exact same way and you have to paste in 4 separate configs.

Reply:
made this change after our discussion